packages/vue-router/lib/vue-router-routing-provider.ts

```diff
@@ -0,0 +1,43 @@
+
+import { onSettle, type RouteResolver, type RoutingProvider, type StartRouteChangeCallback } from '@bugsnag/browser-performance'
+import pathToRegexp from 'path-to-regexp'
+import { type RouteRecordRaw, type Router } from 'vue-router'
+
+function flattenRoutes (routes: RouteRecordRaw[], _prefix: string = ''): string[] {
+  const prefix = `${!_prefix || _prefix === '/' ? _prefix : `${_prefix}/`}`
+  return [
+    ...routes.map(route => `${prefix}${route.path || ''}`),
+    ...routes.reduce<string[]>(
+      (accum, route) => [...accum, ...(route.children ? flattenRoutes(route.children, `${prefix}${route.path}`) : [])],
+      []
+    )
+  ]
+}
+
+export class VueRouterRoutingProvider implements RoutingProvider {
+  router: Router
+  resolveRoute: RouteResolver
+
+  constructor (router: Router, basename?: string) {
```

Comment:
Shall we call this `base` to match the Vue naming? (e.g. https://router.vuejs.org/api/#createWebHistory)